### PR TITLE
Added WhitespacePreTokenizer

### PR DESCRIPTION
**Pull request by jkrukowski**
In this PR:
- added `WhitespacePreTokenizer`
- added pretokenizer tests
- removed `public` from internal class methods

### CODE REVIEW

File: Tests/PreTokenizerTests/PreTokenizerTests.swift [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+//  PreTokenizerTests.swift
+//
+//  Created by Jan Krukowski on 23/11/2023.
+//
+
+import XCTest
+import Hub
+@testable import Tokenizers
+
+class PreTokenizerTests: XCTestCase {
```
Comment (pcuenca) [inline Tests/PreTokenizerTests/PreTokenizerTests.swift:11]: This alone is extremely useful, thanks a lot!

File: Sources/Tokenizers/PreTokenizer.swift [hunk 2]
```diff
@@ -53,6 +54,7 @@ struct PreTokenizerFactory {
         case .Punctuation: return PunctuationPreTokenizer(config: config)
         case .Digits: return DigitsPreTokenizer(config: config)
         case .Split: return SplitPreTokenizer(config: config)
+        case .Whitespace: return WhitespacePreTokenizer(config: config)
         default       : fatalError("Unsupported PreTokenizer type: \(typeName)")
```
Comment (pcuenca) [inline Sources/Tokenizers/PreTokenizer.swift:58]:
```suggestion
        case .Whitespace: return WhitespacePreTokenizer(config: config)
        default: fatalError("Unsupported PreTokenizer type: \(typeName)")
```

Can we take the opportunity to remove this old alignment spacing that is no longer up to date? 🙏

**Review by pcuenca — APPROVED** [review]
Very clean and complete, thank you very much! 🙌

I just suggested an unrelated nit in a line adjacent to yours.

File: Sources/Tokenizers/PreTokenizer.swift (hunk 2 above)
Comment (jkrukowski) [reply Sources/Tokenizers/PreTokenizer.swift]: done

**Review by pcuenca — APPROVED** [review]
Thanks again! 🙌

**Pull request merged**